### DISCUSSION TITLE
What facilities are available for interacting with a hydro component from a normal parent Razor Pages?

### DISCUSSION THREAD

**Discussion by dodyg** [hydrostack/hydro#117] [Q&A]
``` cshtml
@page "/"
@using Hydro
@{
    Layout = "/Pages/Shared/_Layout.cshtml";
    var message = "Hello World";
}

<message text="@message"></message>
```

The documentation focus on component to component interactions but I can't find what kind of facilities are available between host RazorPages with a hydro component. 

In above code we are just passing a Razor variable statically. Is there anything else available?

**Comment by kjeske** [comment]
Could you give an example of what kind of facilities you mean? Razor pages are not interactive, so the only thing you usually do in here is to pass some initial data to the component.

**Comment by dodyg** [comment]
``` cshtml
@page "/"
@using Hydro
@{
    Layout = "/Pages/Shared/_Layout.cshtml";
    var message = "Hello World";
}

<message text="@message"></message>

<button onclick="??">Modify Message</button>
```

Is there any way to interact with hydro component via JavaScript from the host page?

**Comment by kjeske** [comment]
The interactions are possible only inside the components. The host pages are used to render these components.
